src/main/java/pl/akademia/api/promotion/PromotionCodeRestController.java

```diff
@@ -23,20 +25,30 @@ public PromotionCodeRestController(PromotionCodeService promotionCodeService) {
         return new ResponseEntity<>(promotionCodeService.createPromotionCode(activeDays, discount), HttpStatus.CREATED);
     }
 
-    @GetMapping("/promocodes/{a}")
-    public ResponseEntity<List<PromotionCode>> getAllPromotionCodes(@RequestParam(required = false) String a){
-        List<PromotionCode> promotionCodes;
-        if ("active".equals(a)) promotionCodes = promotionCodeService.getActivePromotionCodeByCode();
-        else if ("inactive".equals(a)) promotionCodes = promotionCodeService.getInactivePromotionCodeByCode();
-        else if (a == null) promotionCodes = promotionCodeService.getAllPromotionCode();
-        else return new ResponseEntity<>(HttpStatus.NOT_FOUND);
-        if (promotionCodes.isEmpty()) return new ResponseEntity<>(HttpStatus.NO_CONTENT);
-        return new ResponseEntity<>(promotionCodes, HttpStatus.OK);
+    @GetMapping("/promocodes/{promoCodeStatus}")
+    public ResponseEntity<List<PromotionCode>> getAllPromotionCodes(@RequestParam(required = false) String promoCodeStatus){
+
+        List<PromotionCode> promotionCodes = null;
+        if (promotionCodes.isEmpty()) {
```

Comment:
ten warunek powinien być jeśli już to niżej, bo nigdy w niego nie wpadniemy bo nie ma przypisania listy